h/security/predicates.py

```diff
@@ -217,6 +221,36 @@ def get_authenticated_users_membership():
     )
 
 
+@requires(authenticated_user, group_found)
+def group_member_add(identity, context: AddGroupMembershipContext):
+    assert (
+        context.new_roles is not None
+    ), "new_roles must be set before checking permissions"
+
+    def get_authenticated_users_roles():
+        """Return the authenticated users roles in the target group."""
+        for membership in identity.user.memberships:
+            if membership.group.id == context.group.id:
+                return membership.roles
+
+        return []
+
+    authenticated_users_roles = get_authenticated_users_roles()
+
+    if GroupMembershipRoles.OWNER in authenticated_users_roles:
+        # Owners can add members with any roles.
+        return True
+
+    if GroupMembershipRoles.ADMIN in authenticated_users_roles:
+        # Admins can add members with any role below admin.
+        return (
+            GroupMembershipRoles.OWNER not in context.new_roles
+            and GroupMembershipRoles.ADMIN not in context.new_roles
+        )
+
+    return False
```

Comment:
The `get_authenticated_users_roles()` helper has been duplicated a few times now, I'll send a follow-up PR to refactor and deduplicate this and other duplication in `predicates.py`.